.github/pull_request_template.md

```diff
@@ -0,0 +1,29 @@
+CHANGELOG: `no-impact` / `Added` / `Changed` / `Deprecated` / `Removed` / `Fixed` / `Security`
+
+Describe how these changes affect the customer in one line. This is the thing you are going to add to a changelog.txt file.
+
+> CHANGELOG entries will be pulled directly into the release changelog, so use
+customer-facing language. Start with one of the active verbs above,
+write a full sentence, and end with a period. If these changes do not
+impact the customer, then this line should read CHANGELOG: no-impact.
+
+### Description
+Describe why you're making this change, in plain English.
+
+### Testing & Repro steps
+* In the case of bugs, describe how to replicate
+* If any manual tests were done, document the steps and the conditions to replicate
+* Call out any important/ relevant unit tests, e2e tests or integration tests you have added or are adding
+
+### Links
+Include any links here that might be helpful for people reviewing your PR (GH issues, API docs, external benchmarks, tools docs, etc). If there are none, feel free to delete this section.
```

Comment:
I would like to suggest we start including links to internal systems here if they don't leak customer information. E.g. Asana tickets, consul-enterprise issues, etc. While we should transcribe and perhaps summarize that information in the PR, having the links helps keep things from falling through the cracks and can provide context if something is lost in translation.

Reply:
Maybe a slightly hair-brained idea, but I wonder if there's a convenient way to use our internal URL shortener (which requires SSO) for this kinda thing? Like a Chrome plugin that spots when you're trying to paste an internal link and replaces it with an obfuscated version.

Reply:
I added guidance to internal links and how to use hashi links